### PR TITLE
Introduce binary format for MEDIT files

### PR DESCRIPTION
**Pull request by ChristosT**
This PR introduces support for binary medit files resolving thus #295 .
It is built based on the documentation provided at https://github.com/LoicMarechal/libMeshb which serves as a reference implementation of the MEDIT/GMF format.

Following the suggestion in #299 it reads and writes data in blocks .
Also, it skips fields gracefully as it was proposed in #239 .

The code could be adapted in order to replace the current used for ascii files, but I will be looking into this in a separate PR after I get your feedback about this .

### CODE REVIEW

**Comment by ChristosT** [comment]
the test meshes weren't under git lfs, this should fix this

**Comment by nschloe** [comment]
This looks _very_ nice. @gdmcbain Would you like to have a look as well?

File: meshio/medit/_medit.py [hunk 1]
```diff
@@ -171,4 +331,148 @@ def write(filename, mesh, float_fmt=".15e"):
...
+
+        version = 3
+        itype = "i4"
+        postype = "i8"
+        ftype = "f8"
+        # according to manual keywords are always written as i4 independently of
+        # the file version
+        keytype = "i4"
+
+        # if we store internaly 64bit integers upgrade file version
```
Comment (gdmcbain) [inline meshio/medit/_medit.py:345]: spell 'internally'

File: meshio/medit/_medit.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 <https://www.ljll.math.upmc.fr/frey/publications/RT-0253.pdf>
 <https://www.math.u-bordeaux.fr/~dobrzyns/logiciels/RT-422/node58.html>
 for something like a specification.
```
Comment (gdmcbain) [inline meshio/medit/_medit.py:8]: Are the old references rendered obsolete by libMeshb?  Should they be deleted?

**Comment by gdmcbain** [comment]
It does look very nice, doesn't it!  I had a quick glance on Tuesday and quickly concluded that @ChristosT knows a lot more about MEDIT than I do!  (I hadn't even heard of GMF.)  It's a most welcome contribution.

I don't have ready access to my own old MEDIT cases at the moment, but I've stepped through some basic conversions in pdb; as it's also passing the suite, including its own good new additions, I'd recommend merging.

File: meshio/medit/_medit.py (hunk 1 above)
Comment (ChristosT) [reply meshio/medit/_medit.py]: Done. thank you

File: meshio/medit/_medit.py (hunk 2 above)
Comment (ChristosT) [reply meshio/medit/_medit.py]: Yes the old ones are just a subset.  So I will be keeping only the latest

**Comment by ChristosT** [comment]
Thank you for your kind words and the review !

**Pull request merged**